specs/monitoring_defs.adoc

```diff
@@ -0,0 +1,105 @@
+= Flow Framework Refactoring
+
+== Introduction
+
+Current working procedure:
+
+* Any service/agent defining and implementing a flow pushes the corresponding
+  flow definitions to its specific definitions etcd directory.
+* The Agent/service flow framework reads its specific definition files and
+  does all validations and executes the flow.
+
+ex:
+
+* Node-agent pushes tendrl_definitions_node_agent to
+  /tendrl_definitions_node_agent in etcd, gluster integration pushes
+  tendrl_definitions_gluster_integration to
+  /tendrl_definitions_gluster_integration and so on...
+* node-agent is a service that reads flow definitions from
+  /tendrl_definitions_node_agent in etcd and gluster-integration is a process
+  that reads /tendrl_definitions_gluster_integration from etcd and "currently"
+  each of them replicates the flow definition parsing logic.
+
+== Problem description
+
+Any component requiring to define its own flows will then need to implement
+(currently followed)/extend(from a generic one in tendrl/common) the flow
+framework and this requires the component to be run as a service as the flow
+framework in each of these component will only and always read only its
+specific definition and whenever there is a job (intended for it) in the etcd
+queue.
+
+== Use Cases
+
+* Monitoring code cannot be part of any component of tendrl core stack.
+* The monitoring stack needs to provide a flow for configuring the monitoring
+  agents maintained by tendrl.
+
+== Proposed change
+
+* Push all flow definitions to a path under a common root directory say for
+  ex: /tendrl_definitons/ in etcd
+* Now, when there's a job in the queue, the flow framework will make use of
+  either of the following:
+	** A part of the full qualified package name in the run parameter of job
+	** A separate field to indicate the definitions namespace(i.e, the name
+	   under the /tendrl_definitions/ where the definition can be obtained)
+       to read definitions from appropriate etcd directory.
+  to create the definitions path instead of hard-coding it.
+* And from there the already existing flow framework as usual will extract and
+  execute the pre, post and the normal flow atoms.
+
+=== Alternatives
+
+* Append monitoring related definitions yaml to node_agents definitions
+  directory in etcd.
+
+    ** Drawbacks
```

Comment:
I dont think you need tabs if you are using ```**``` and ```***```. They automatically would happen in final doc